src/time_machine/cli.py

```diff
@@ -82,7 +117,11 @@ def migrate_contents(contents_text: str) -> str:
     except SyntaxError:
         return contents_text
 
-    callbacks = visit(ast_obj)
+    # Phase 1: Collect all information
+    state = collect_migration_state(ast_obj)
+
+    # Phase 2: Generate callbacks based on collected state
+    callbacks = generate_callbacks(state)
```

Comment:
I don't like this two-pass approach, it decouples things too much

it's better to have the token-rewriting functions read bits of state as necessary

see examples in django-upgrade such as 

https://github.com/adamchainz/django-upgrade/blob/37f31275abfe9b9d6e56627adbfb1a9ac12674ae/src/django_upgrade/fixers/admin_register.py#L25